test/EngineTest.cpp

```diff
@@ -23,114 +32,259 @@ auto I = [](const auto& id) {
+  // without arguments.
+  normalJoinTest(const IdTable& LeftIdTable,
+      const size_t LeftJC,
+      const IdTable& RightIdTable,
+      const size_t RightJC,
+      const IdTable& SampleSolution,
+      const bool TestForSorted = false): leftIdTable(LeftIdTable.clone()),
+      leftJC(LeftJC),
+      rightIdTable(RightIdTable.clone()),
+      rightJC(RightJC),
+      testForSorted(TestForSorted),
+      sampleSolution(SampleSolution.clone())
+  {}
+};
 
-  for (size_t i = 1; i <= 10000; ++i) {
-    a.push_back({I(4 + i), I(2 + i)});
+/*
+ * @brief Tests, if the given IdTable has the same content as the sample
+ * solution and, if the option was choosen, if the IdTable is sorted by
+ * the join column.
+ *
+ * @tparam TABLE_WIDTH Just ignore it. Made to be infered from the function
+ *  paramters.
+ *
+ * @param table The IdTable that should be tested.
+ * @param sampleSolution The sample solution. Doesn't need to be sorted,
+ *  or the same order of rows as the table.
+ * @param testForSorted If this is true, it will also be tested, if the table
+ *  is sorted by the join column.
+ * @param jc The join column of the table.
+ * @param t Ignore it. It's only here for being able to make better messages,
+ *  if a IdTable fails the comparison.
+*/
+void compareIdTableWithSolution(const IdTable& table, 
+    const IdTable& sampleSolution,
+    const bool testForSorted = false,
+    const size_t jc = 0,
+    ad_utility::source_location t = ad_utility::source_location::current()
+    ) {
+  // For generating more informative messages, when failing the comparison.
+  std::stringstream traceMessage;
+  traceMessage << "compareIdTableWithSolution comparing IdTable\n";
+  for (size_t row = 0; row < table.size(); row++) {
+    for (size_t column = 0; column < table.numColumns(); column++) {
+      traceMessage << table(row, column) << " ";
+    }
+    traceMessage << "\n";
+  }
+  traceMessage << "with IdTable \n";
+  for (size_t row = 0; row < sampleSolution.size(); row++) {
+    for (size_t column = 0; column < sampleSolution.numColumns(); column++) {
+      traceMessage << sampleSolution(row, column) << " ";
+    }
+    traceMessage << "\n";
+  }
+  auto trace = generateLocationTrace(t, traceMessage.str());
+ 
+  // Because we compare tables later by sorting them, so that every table has
+  // one definit form, we need to create local copies.
+  IdTable localTable = table.clone();
+  IdTable localSampleSolution = sampleSolution.clone();
+
+  // Do the IdTable and sampleSolution have the same dimensions?
+  ASSERT_EQ(localTable.size(), localSampleSolution.size());
+  ASSERT_EQ(localTable.numColumns(), localSampleSolution.numColumns());
+
+  if (testForSorted) {
+    // Is the table sorted by join column?
+    auto oldEntry = localTable(0, jc);
+    for (size_t i = 1; i < localTable.size(); i++) {
+      ASSERT_TRUE(oldEntry <= localTable(i, jc));
+      oldEntry = localTable(i, jc);
+    }
```

Comment:
The current version ([ec21f8f](https://github.com/ad-freiburg/qlever/pull/821/commits/ec21f8fe96e7e8a1e80e189df2c0d3d9042cb301)) already uses `std::ranges::is_sorted`, but in a much more verbose way `std::ranges::is_sorted(localTable, {}, [joinColumn](const auto& row){return row[joinColumn];})`. So this would actually improve readability quite a bit.

Only a small problem with the idea: `getColumn` is a private function, not a public one. And I'm not sure enough in my knowledge about the way tables are build, to say, if the change to public would be a bad idea.